### ISSUE TITLE
Undo does not work

### ISSUE THREAD

**Issue by michaelfresco** [kjnh10/ExcelLikeVim#4]
What a cool plugin!

I think that `u` does not work properly. When I delete a row with `dd` and then press `u` you'd expect it to undo the most recent delete. Is undo not implemented yet? Or do I really miss something basic here.

Cheers

**Comment by kjnh10** [comment]
Thanks for telling!

Sorry, as you guess, the undo feature hasn't been implemented yet.

I'm thinking to implement, though I haven't come up with the solution of how to do it because excel application does not have the undo feature for the action invoked by VBA function.

It may be the solution that when a function invoked, putting the reverse function of it to a stack and calling a function from the stack when doing undo. But it seems complex..

For now, I will modify the README to note that undo feature is missing.

**Comment by BenKettlewell** [comment]
This makes me sad. Even ctrl+z doesn't undo which makes this awesome tool pretty much unusable. \sigh\

**Comment by CaptNapalm** [comment]
Yeah, I was enjoying this tool right up to the point that I deleted a line by accident and wasn't even able to use the default mouse click undo or ctrl-z. Really needs a solution in order to be usable. When I'm editing things in vim I'm used to deleting stuff all the time and then just pressing 'u'.

**Comment by kjnh10** [comment]
@CaptNapalm 
Sorry for reply.

The fundamental modification seems difficult to me as I explained above. Sorry.

My work around is
・save file frequently
・'dd' saves the line to a kind of regisiter, so you can recover the line by pushing 'p' key.

**Comment by analyticd** [comment]
Regarding 'dd' followed by 'p' to recover a value from a "kind of register". In my case you may be interested to know that I deleted one english word on a line and moved to a new line/row where I pasted via 'p' and it, suprisingly, pasted a row with several columns of Japanese text. Almost like something from your environment perhaps?

**Comment by kjnh10** [comment]
@analyticd 
Thanks for bug reporting. Much appreciated.

I maybe fixed the bug in the hotfixes/yank_paste branch. (not released yet)
Could you check out it and test it?

**Comment by analyticd** [comment]
@kjnh10 I checked out the hotfixes/yank_paste branch and tested. You seem to have fixed the paste of Japanese text issue. 

As a side note: 

For some reason 'dd' doesn't seem to work for me, and thus 'y' yank has no effect subsequently, in 2.3.4 or in the hotfixes/yank_paste branch. 'O' open line above worked fine still.

**Comment by analyticd** [comment]
Possibly an important thing for you to know is that when I first start up a spreadsheet when vimx is enabled as an add-in and I press say, 'j', I get a message about reopening the registry.(not sure the suffix here sorry) and then a pop up dialog that has a message: 'couldn't get newkey'. If I click 'ok' in that dialog then I can proceed, but only some of vimx features work as noted above.

**Comment by analyticd** [comment]
However, if I select a line(s) in visual mode and then delete via 'd' and subsequently paste via 'y', that works fine.

**Comment by kjnh10** [comment]
strange...

could you open VB Editor and execute 'n_dd' sub in 'vimize' module?
At first, I want to know this sub fails or keystroke 'dd' fails to connect to 'n_dd' sub.

Anyway thanks for investigating.

**Comment by analyticd** [comment]
Ok,  I navigated to a cell using vimx navigation, entered a number in a cell and left my cursor there. On the Developer tab I then created a macro (using 'Macros' button) and typed 'vimize.n_dd' and then ran it. It succeeded in deleting the cell. So that is good. I subsequently tried to do the same with 'dd' but it pauses for a while and then does nothing. So at least you know more now.

I noticed when vimize.n_dd succeeds it throughs the cursor location back to the first column. I wonder if you could track the cell location and at all times and then return to that cell after a 'dd' or other side-effecting action. 

For now, as I said earlier, at least visual mode allows for same functionality as 'dd', so that is nice.

Once you figure out undo, this will be really great. It is already really useful.

**Comment by kjnh10** [comment]
Thanks,

I have few more questions.
(1): 'gg' works as jumping to first row. (this is to confirm to continuous key stroke (not single key )is working) 
(2): paste is via 'p' not 'y'? or are you using costomize file (configure.bas)?ll

Regarding undo feature, this seems difficult to implement as I described on top of this therad....

**Comment by analyticd** [comment]
1) 'gg' does not work
2) Sorry I wasn't clear. 'y' to copy and 'p' to paste work fine.

Re undo: yes I saw that you said that, I was just offering encouragement to undertake the task (unless it is too much of course). 

In your use of vimx, how do you workaround not having undo? If you share this information, perhaps we can adopt similar approach to mitigate undo issues.

**Comment by snth** [comment]
I also just hit the Undo issue about 5 minutes into my test.

I understand that it might be difficult to implement `u` as Undo if Excel doesn't expose this on the VBA side but why does Ctrl+Z not work anymore with ExcelLikeVim enabled?

**Comment by magnusriga** [comment]
Same issue here. Would it be possible to still keep the ctrl and alt shortcuts from standard Excel?
